packages/shared/src/components/squads/SharePostBar.tsx

```diff
@@ -30,23 +31,42 @@ function SharePostBar({
   onNewSquadPost,
   disabled = false,
 }: SharePostBarProps): ReactElement {
+  const { displayToast } = useToastNotification();
   const [url, setUrl] = useState('');
   const onSharedSuccessfully = () => setUrl('');
-  const { mutateAsync: getPost } = useMutation(getPostByUrl, {
-    onSuccess: (post) => {
-      onNewSquadPost({ post, onSharedSuccessfully });
-    },
-    onError: (err: ApiErrorResult, link) => {
-      if (
-        link === '' ||
-        allowedSubmissionErrors.includes(
-          err?.response?.errors?.[0].extensions.code,
-        )
-      ) {
-        onNewSquadPost({ url, onSharedSuccessfully });
-      }
+  const { mutateAsync: getPrivateLink, isLoading: isLoadingPreview } =
+    useMutation(getExternalLinkPreview, {
+      onSuccess: (preview, link) => {
+        const privateLink = { url: link, ...preview };
+        onNewSquadPost({ privateLink, onSharedSuccessfully });
+      },
+      onError: (err: ApiErrorResult, link) => {
+        const rateLimited = hasApiError(err, ApiError.RateLimited);
+        if (rateLimited) {
+          displayToast(rateLimited.message);
+        }
+
+        onNewSquadPost({ privateLink: { url: link }, onSharedSuccessfully });
+      },
+    });
+  const { mutateAsync: getPost, isLoading: isCheckingUrl } = useMutation(
```

Comment:
This is actually also re-used, maybe can go in the same hook or the existing squad hook?

Reply:
I have thought of doing it but refrained while on it as it felt not that much difficult, but now that you mentioned it, let's see how it will look like.